components/InputWithClearIcon.tsx

```diff
@@ -15,7 +15,7 @@ export default function InputWithClearIcon(props: InputWithClearIconProps) {
             onChange={(event) => props.onChange(event.target.value)}
         />
         {props.value && (
-            <IconX className='absolute right-2 top-1/2 transform -translate-y-1/2 cursor-pointer text-gray-500' onClick={() => props.onChange("")} />
+            <MemoIconX className='absolute right-2 top-1/2 transform -translate-y-1/2 cursor-pointer text-gray-500' onClick={() => props.onChange("")} />
```

Comment:
memo doesn't do anything if the render create a new reference in the onClick. For it to take effect we need to use an useCallback for the function